scripts/deployAll.ts

```diff
@@ -0,0 +1,236 @@
+import { BytesLike } from "ethers";
+import hre from "hardhat";
+const { web3, ethers } = hre;
+
+async function main() {
+  const [deployer] = await ethers.getSigners();
+  const deployerAddress = deployer.address;
+  console.log(`\n\n\n Deployer Address: ${deployerAddress} \n\n\n`);
+
+  if (hre.network.name === "mainnet") {
+    console.log("\n\n Deploying Contracts to mainnet. Hit ctrl + c to abort");
+  } else if (hre.network.name === "kovan") {
+    console.log("\n\n Deploying Contracts to kovan...");
+  } else {
+    console.log("\n\n Deploying Contracts to", hre.network.name, "...");
+  }
+
+  const InstaIndex = await ethers.getContractFactory("InstaIndex");
+  const instaIndex = await InstaIndex.deploy();
+  await instaIndex.deployed();
+
+  const instaIndexAddress = instaIndex.address;
+  console.log("instaIndex deployed: ", instaIndexAddress);
+
+  const InstaList = await ethers.getContractFactory("InstaList");
+  const instaList = await InstaList.deploy(instaIndexAddress);
+  await instaList.deployed();
+
+  console.log("instaList deployed: ", instaList.address);
+
+  const InstaAccount = await ethers.getContractFactory("InstaAccount");
+  const instaAccount = await InstaAccount.deploy(instaIndexAddress);
+  await instaAccount.deployed();
+
+  console.log("instaAccount deployed: ", instaAccount.address);
+
+  const InstaConnectors = await ethers.getContractFactory("InstaConnectors");
+  const instaConnectors = await InstaConnectors.deploy(instaIndexAddress);
+  await instaConnectors.deployed();
+  console.log("instaConnectors deployed: ", instaConnectors.address);
+
+  const InstaEvent = await ethers.getContractFactory("InstaEvent");
+  const instaEvent = await InstaEvent.deploy(instaList.address);
+  await instaEvent.deployed();
+
+  console.log("instaEvent deployed: ", instaEvent.address);
+
+  const InstaMemory = await ethers.getContractFactory("InstaMemory");
+  const instaMemory = await InstaMemory.deploy();
+  await instaMemory.deployed();
```

Comment:
Also in here, we are repeating a lot of code again and again!
Why don't we create a deployContract function!